src/mame/igs/igs_fear.cpp

```diff
@@ -45,42 +50,75 @@ class igs_fear_state : public driver_device
 	required_device<palette_device> m_palette;
 	required_region_ptr<uint8_t> m_gfxrom;
 
+	required_device<ticket_dispenser_device> m_ticket;
+	required_ioport_array<2> m_io_dsw;
+	required_ioport_array<2> m_io_trackball;
+
 	void main_map(address_map &map);
 
 	void sound_irq(int state);
 	void vblank_irq(int state);
 
-	void draw_sprite(bitmap_ind16 &bitmap, const rectangle &cliprect, int xpos, int ypos, int height, int width, int palette, int romoffset);
+	void draw_sprite(bitmap_ind16 &bitmap, const rectangle &cliprect, int xpos, int ypos, int height, int width, int palette, int flipx, int romoffset);
 
 	uint32_t screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
+
+	uint32_t m_gpio_o;
```

Comment:
We prefer to keep function signatures and variables separate.  I usually do all the function stuff and then all the variables, but the other way is fine too if that works better for you.  I used to write stuff all mixed together like this and it turns out it wasn't great for maintaining it later.